internal/message/amqp/interface.go

```diff
@@ -0,0 +1,44 @@
+package amqp
+
+import (
+	"io"
+
+	"github.com/streadway/amqp"
+)
+
+// default implementation leverages the real "streadway/amqp" dialer
+var defaultDialerAdapter DialerAdapter = func(url string) (Connection, error) {
+	conn, err := amqp.Dial(url)
```

Comment:
Is there a default timeout associated with this? Just thinking about the connection retry delay of 5s vs. connect timeout.

Reply:
`amqp.Dial` calls https://pkg.go.dev/github.com/streadway/amqp#DialConfig which has a hard-coded `defaultConnectionTimeout` of 30s. This will, however, only hang for that amount of time if the endpoint exists. So, it could hang for say, 150 secs, before complete failure.

what are you thinking here exactly?

Reply:
I guess I was thinking the connect timeout & ticker would interfere, but it seems like Ticker will "catch-up" instead of losing the intermediate ticks while the connection is ongoing.

Reply:
Yep. I was able to prove that really quick with a test so I think we're good here.